src/plugins/timedLightTheme.tsx

```diff
@@ -0,0 +1,186 @@
+        <Forms.FormTitle>Preferred Themes</Forms.FormTitle>
+        {canUseClientThemes()
+            ? <Forms.FormText variant="text-md/bold">
+                Light: {snakeToTitleCase(light === -1 ? "default" : Themes[light])} -
+                Dark: {snakeToTitleCase(dark === -1 ? "default" : Themes[dark])}
+            </Forms.FormText>
+            : <Forms.FormText variant="text-md/bold">
+                Can't use nitro client themes. Restart Discord if this is wrong
+            </Forms.FormText>
+        }
+    </>;
+}
+
+const settings = definePluginSettings({
+    start: {
+        description: "When to enter light mode (24-hour time)",
+        type: OptionType.STRING,
+        default: "08:00",
+        placeholder: "xx:xx",
+        isValid: t => /^\d{0,2}(?::\d{0,2})?$/.test(t),
+        onChange: () => Vencord.Plugins.isPluginEnabled("TimedLightTheme") && (Vencord.Plugins.plugins.TimedLightTheme as any).checkForUpdate(),
+    },
+    end: {
+        description: "When to enter dark mode (24-hour time)",
+        type: OptionType.STRING,
+        default: "20:00",
+        placeholder: "xx:xx",
+        isValid: t => /^\d{0,2}(?::\d{0,2})?$/.test(t),
+        onChange: () => Vencord.Plugins.isPluginEnabled("TimedLightTheme") && (Vencord.Plugins.plugins.TimedLightTheme as any).checkForUpdate(),
+    },
+    preferredTheme: {
+        description: "Store which theme to use for light and dark mode",
+        type: OptionType.COMPONENT,
+        component: DisplayThemeComponent
+    },
+});
+
+export default definePlugin({
+    name: "TimedLightTheme",
+    authors: [Devs.TheSun],
+    description: "Automatically enables/disables light theme based on the time of day",
+    settings,
+
+    checkForUpdate() {
+        const { start, end } = settings.store;
+        if (!start || !end)
+            return showError();
+
+        const startTimestamp = toAdjustedTimestamp(start);
+        const endTimestamp = toAdjustedTimestamp(end);
+        if (startTimestamp >= endTimestamp)
+            return showError();
+
+        const now = Date.now();
+
+        const themeId = getClientThemeId();
+        if (!settings.store.preferredTheme) {
+            const theme = getBasicTheme();
+            settings.store.preferredTheme = {
+                light: theme === "light" ? themeId : -1,
+                dark: theme === "dark" ? themeId : -1
+            };
+        }
+
+        if (now < startTimestamp) {
+            updateThemeIfNecessary("dark");
+            nextChange = setTimeout(() => this.checkForUpdate(), startTimestamp - now);
+        }
+        else if (now >= startTimestamp && now <= endTimestamp) {
+            updateThemeIfNecessary("light");
+            nextChange = setTimeout(() => this.checkForUpdate(), endTimestamp - now);
+        }
+        else if (now > endTimestamp) {
+            updateThemeIfNecessary("dark");
+            nextChange = setTimeout(() => this.checkForUpdate(), (startTimestamp + 86400_000) - now);
```

Comment:
could you add a comment stating how much time this is? 
```suggestion
            nextChange = setTimeout(() => this.checkForUpdate(), (startTimestamp + 86_400_000) - now);
```